crates/cxx-qt-gen/src/parser/property.rs

```diff
@@ -162,25 +189,46 @@ impl ParsedQProperty {
                     ))
                 }
 
-                if let Some(read) = read {
+                let cxx_name_str = if let Some(name) = cxx_name {
+                    name.to_string()
+                } else if rust_name.is_none() {
+                    ident.to_string().to_case(Case::Camel)
+                } else {
+                    ident.to_string()
+                };
+                let mut name = Name::new(ident).with_cxx_name(cxx_name_str);
+
+                if let Some(ident) = rust_name {
+                    name = name.with_rust_name(ident);
+                }
+
+                if read_required {
+                    if let Some(read) = read {
+                        Ok(Self {
+                            name,
+                            ty,
+                            flags: QPropertyFlags {
+                                read,
+                                write,
+                                notify,
+                                reset,
+                                is_final,
+                                constant,
+                                required,
+                            },
+                        })
+                    } else {
+                        Err(Error::new(
+                            punctuated_flags.span(),
+                            "If any flags are passed, READ must be explicitly specified!",
+                        ))
+                    }
+                } else {
```

Comment:
some code comments through this part might be useful to state what each if statement is doing

Reply:
Done, and that naming logic has been extracted, so should be more readable